### ISSUE TITLE
Interface has some barriers to entry. Printing and some menu stuff doesn't work [Windows 10] [bugs]

### ISSUE THREAD

**Issue by jrcharney** [Zettlr/Zettlr#324]
## Description
I like Zettlr, but some of the things either don't work or they are put somewhere that new users might not be used to.

## Reproducing
I recently decided to move from [Notable](https://github.com/notable/notable) to Zettlr because Notable doesn't have printing functionality like Zettlr has.

I backed up my work first in a text file in Notepad++, but then I though I should paste this into a Markdown editor and print it. I considered using Notable at first, but then I noticed it didn't have any print features. So I looked up Zettlr thinking this would be like Notable but with the missing features.

However, my first impressions of it seem to have me trip over a lot of things and there there appears to be plenty of problems. And not printing is just one of them. 

I can't save files unless I rename the file first, but Zettlr will let me make a Project Folder which apparently doesn't help me save the file I want to create but sure like to create a who bunch of empty markdown files in that project folder that I created. (Probably because I may have clicked on the "new file" icon trying to figure out "where's the new file". The feather as a placeholder looks nice, but a blinking cursor would be even better.

Initially I had installed Zettlr from Chocolatey, but the choco package hasn't been updated since version 1.3.0, and the app suggested I upgrade to 1.4.1, which I did. (I know that is the responsibility of the "third party" that maintains the package. You may want to get in contact with that person.  For security reason, you should be the one maintaining the packages on all platforms.)

Before I started doing any work in Zettler, I wanted to take a look an the options.

It appeared it wanted me to install Pandoc and some TeX software (I chose MiKTeX) wasn't installed which this software recommended that I do (even though I didn't realize that KaTeX was part of this program, like Notable) and that I forgot that KaTeX will won't support the `align*` environment but will support it if I use `aligned` instead in my `\begin{}` and `\end{}` tags. So I got over that issue by changing `\begin{align*}` and `\end{align*}` to `\begin{aligned}` and `\end{aligned}`. So that problem solved.  (I wonder if I should uninstall pandoc and MiKTeX if Zettler is supposed to interpret Markdown and LaTeX already? An answer would be nice.)

Another thing I noticed is that when I was trying to paste text from some other source, if I did a straight paste (Ctrl + V), I'd get this blob of HTML despite there being no HTML in it and it was just regular text I wanted to paste into it. So when I told it to just paste the unformatted text (which the command should be a Ctrl + Shift + V), only one line of text is pasted depending on what app you are pasting the text from. I was working on a [MathBin post](http://mathb.in/36237) where there was plenty of Markdown and LaTeX to process.

The first time I inserted text, I got some prompt asking if I would like to save now or later. (I would have though Zettlr would do what Notepad++ and Notable do and stick anything into some temp file until I decide to save it as something else or delete it.) This prompt hasn't shown up since I created a project folder which I though would have been used as a directory to store files. (It stores files, just not the files you want and definitely a lot of timestamped empty markdown files.)

When I click "Toggle readablity mode" it highlights all the non-LaTeX text. (That's not exactly what I was thinking about when I wanted to see what it would look like in a more readable format.) Wouldn't it make more sense to apply some accessibility features with that? (Larger text, text-to-speech, etc.)

Another thing I noticed is that sometimes menu items will not execute. Save, Print, and Quit especially. I couldn't even use the keyboard shortcuts. In fact, when I finally had enough of Zettlr not doing even those things, I had to use the Task Manager to force the application to quit.

## Expected behavior
I get now that I have to create a "Project folder" to store files that I want to save, but the clarity of this requirement needs a better explanation for new users.  I was thinking that I could just make a folder to stick all my markdown files I created into a folder (preferably one that backs stuff up into the cloud like with OneDrive or Google Docs) and done.  That was only part of the thing I needed to do.

Apparently when files are created, they are created with timestamps as filenames. It seems good in concept until you start to think "what are all these empty markdown files here?" So now I have to clean that up.  I didn't realize that this could be done with that rename file feature in the title bar. (It probably would be better if it appears when you hover over a file in the file list. Also have the remove file and share file options near by there instead instead of in the title bar, which is a dangerous place to put a delete file icon.

I was confused by how search work. Apparently the search box is for the file directories and the files in the file list, not the actually text in the file which the other magnifying glass icon is for. So the file search field probably should be moved into sidebar.

Paste should let me paste text. If Zettlr wants to format text, Ctrl+V should do it that, but if I want to paste the raw text, Ctrl+Shift+V.

Some Menu items aren't working, I blame some kind of bug with Electron.

I tried to print something and got a message about how pandoc couldn't be found, never mind that it is installed and there was a full system restart for good measure.

Markdown headers still show the octothorpes (`#`) in headers, asterisks in bold and italic text (What happened to "Format in place"?). Also, links with parenthesis in them do not parse well unless you go back and replace the `(` and `)` with `%28` and `%29`. The parser needs to convert special characters in links to their ASCII compatible code.  Parsing needs work. If anything there should be an option to toggle the appearance of markdown syntax.

A reminder that if you are using `\begin{align*}` and `\end{align*}` in LaTeX, changing `align*` to `aligned` would be nice. (This seems more like something the KaTeX folks should be working on, or better yet they should allow for `align*` to be used as that is standard LaTeX.)

The name of the file created should be automatically generated/updated if the first line is an H1 header (Single `#` followed by text). I'm pretty sure that's how Github does it. It's how Notable does it. And I think that's also how that one app on MacOS does it. (I forget the name of that app.)  Empty markdown files should not be created. They should be created when text is inserted into them. (I now have like 90+ empty files I need to delete. The part that really sucks is I can't select more than one and delete them all a once.)

Rename and Remove files should probably go in that sidebar, as should the Share/Export feature rather than appearing in the title bar. (It's good that it is not enabled when there is no text in the file, but it needs to be in a better place.)

Markdown Text Formats work. Using it is pretty straightforward, though the formatting could use a little work.  The way tables are constructed is good even if LaTeX can't be embedded into cells. (That's more of Markdown's fault. Also, I think you can create tables with LaTeX.) One thing that needs to be improved upon is the appearance of blockquotes (Text starting with `>`), links, bold, and italic text are hard to see in dark mode.  A palette adjustment could help. I like that there is an option for adding CSS styles but loading and creating custom CSS themes would be ideal.

I wasn't quite sure if the "Table of Contents" feature was working. I was thinking there would be something like how Microsoft Word has a sidebar. But I get it now, so that's OK.

## Platform
 - OS: Windows
 - OS Version: 10
 - Zettlr Version: 1.4.1

## Additional information
TLDR: There are a lot of good idea with this application, but it may help to tell the users were a few things are first and to ask some questions as to where they want to put files before they start using this app.  Otherwise it looks beautiful.

Hopefully that pandoc/Print thing can be fixed.

Thanks for reading this bug report.

**Comment by nathanlesage** [comment]
Thank you for your input. We're currently en route to fixing some of these things. It's quite a bit, so we'll need some time to go over it.

Related: #268

**Comment by philiptruax** [comment]
Wow. This is a lot.
@nathanlesage 
* I can confirm the chocolatey package has some weirdness and doesn't update properly. I'll ping the guy maintaining the choco package to update for now unless you want to create an official choco package going forward. 

* I can confirm that a straight copy and paste from that site renders in html in a new document. It also causes the app to hang while trying to render. It's posting the html contents of the clipboard instead of the text contents. 

* I actually like the rename feature where it's at especially in minimal mode, but it did take me a minute to figure out where it was as well. 

* I like the idea for the search bar being moved to the sidebar since I agree that it's confusing that it doesn't search the text and that move would resolve some of that confusion. 

* The app definitely hangs or prevents printing when entering the print screen. I get the following from the console when clicking on the print button with the console open:
>C:\Program Files\Zet…-print-window.js:90 Uncaught DOMException: Blocked a frame with origin "file://" from accessing a cross-origin frame.
    at HTMLSpanElement.<anonymous> (C:\Program Files\Zettlr\resources\app.asar\print\zettlr-print-window.js:90:24)
    at HTMLSpanElement.dispatch (file:///C:/Program%20Files/Zettlr/resources/app.asar/node_modules/jquery/dist/jquery.js:5237:27)
    at HTMLSpanElement.elemData.handle (file:///C:/Program%20Files/Zettlr/resources/app.asar/node_modules/jquery/dist/jquery.js:5044:28)

* If I don't refresh the page I can get to the print options only on the first time I click, but I can't close the window. If I have dev tools open I can refresh and close the window, however the printer options window never opens. 

* I also volunteer to create some video tutorials when I get some time if this would be helpful. Might take me couple of weeks to get the cycles for this though. 

@jrcharney 
* The formatting in place does actually happen however it still leaves the characters visible. If you want to change this behavior for the octothorpes turn on the option "Hide Heading Characters" in the display tab in options. This will still display the heading size next to which you can click on to change the heading size, but it won't hide formatting options entirely. I don't believe there's an option to hide the rest of the characters in the current product.

* You can remove the automatic filenames by ID in the advanced options. If you're looking for a blank filename go into the options under the Advanced tab and remove the default %id.md from the default Pattern for new filenames area. To actually create the file requires clicking enter after the name appears in the top left for me.  I think maybe some video tutorials or a first run guide might be useful for this.

* As a workaround for the empty files, go to the folder containing them on your hard drive and you'll be able to multi-select from there. The app will eventually update on it's own to reflect that or a restart will do it immediately. You can also externally edit the files in another program and Zettlr will see the updates, just make sure the option under General tab "Always load remote changes to the file" is toggled.

**Comment by nathanlesage** [comment]
> I can confirm the chocolatey package has some weirdness and doesn't update properly.

Yeah, I assume chocolatey doesn't have this nice mechanism such as homebrew, which automatically updates once a new GitHub release is online. I explicitly don't support the cocolatey-release, because I don't have the time to maintain more than what I'm offering on the site. I am happy to link to these resources, but I can't take care of these, unfortunately.

> It's posting the html contents of the clipboard instead of the text contents.

That is by design, because the main target group of Zettlr are academics who know Word. And what does Word do normally? Retain formatting. Therefore you'll need to additionally hold I think Alt (?) to get the prompt from Word to paste only the text contents. There's this `<meta>`-tag, which I need to filter out, definitely!

> I actually like the rename feature where it's at especially in minimal mode, but it did take me a minute to figure out where it was as well.

We're currently beginning to assess and redesign huge parts of the app due to these things, there are one designer and a UX guy working on this right now :)

> I like the idea for the search bar being moved to the sidebar

I think this is already on the list!

> The app definitely hangs or prevents printing when entering the print screen.

Ah, this is simply an electron security mechanism they seem to have activated on Electron 6. I'll look into it.

> I also volunteer to create some video tutorials when I get some time if this would be helpful.

If you'd like to, please go forward! :)

**Comment by nathanlesage** [comment]
Soooo, as the print window is fixed now, I'm going to close this issue as it's not narrow enough. In case you believe some things are still issues which you'd like to change, and there are no open issues adressing this, feel free to open certain issues with single requests.
